swift/common/middleware/proxy_logging.py

```diff
@@ -349,6 +409,19 @@ def __call__(self, env, start_response):
 
         self.mark_req_logged(env)
 
+        if self.perfdata:
+            add_perfata = False
+            if not self.perfdata_user_agents:
+                add_perfata = True
+            else:
+                req = Request(env)
```

Comment:
`env.get('HTTP_USER_AGENT')` may be quicker, and avoid creating a `Request` object twice.

Reply:
Ok, it's better, but there is a catch: if the user agent is None, an exception will be raised.